include/real/exact_number.hpp

```diff
@@ -693,190 +694,322 @@ namespace boost {
                     throw exponent_overflow_exception();
                 }
 
-                min_boundary_n.digits = {1};
-                min_boundary_n.exponent = -1 * (maximum_precision);
-                min_boundary_n.positive = false;
-
-                min_boundary_p.digits = {1};
-                min_boundary_p.exponent = -1 * (maximum_precision);
-
-                T base = (std::numeric_limits<T>::max() /4)*2 - 1;
-                T H = base/2 + 1;
-                half.digits = {H};
-                half.exponent = 0;
+                exact_number<T> zero = exact_number<T> ();
+                if(divisor == zero){
+                    throw divide_by_zero();
+                }
 
-                // we ignore exponents, then set them in the end.
-                // (a * 10^x) / (b*10^y) = (a*10 / b*10) * 10^((x-1)-(y-1))
-                // 100/20 -> 1/2 * (10)
-                int exponent_dif = (numerator.exponent - 1) - (divisor.exponent - 1);
+                /* Special Cases */
+                if((*this) == zero){
+                    return;
+                }
 
-                numerator.exponent = 1;
-                divisor.exponent = 1;
-                tmp.digits = {1};
-                tmp.exponent = 1;
+                exact_number<T> one;
+                one.digits={1};
+                one.exponent=1;
+                if(divisor == one){
+                    return;
+                }
 
-                exact_number<T> zero = exact_number<T>(); 
+                bool positive = (this->positive == divisor.positive);
 
-                if (divisor == tmp) {
-                    this->exponent = exponent_dif + 1;
+                exact_number<T> minus_one = one;
+                minus_one.positive = false;
+                if(divisor == minus_one){
                     this->positive = positive;
                     return;
                 }
 
-                if (divisor == (*this)) { 
-                    (*this) = tmp;
-                    (*this).positive = positive;
+                if(divisor == (*this)){
+                    (*this) = one;
                     return;
                 }
 
-                if ((*this) == zero)
-                    return;
-                    
-                if (divisor == zero)
-                    throw(boost::real::divide_by_zero());
-
-                // N < D --> 0 < abs(Q) < 1
-                if (numerator < divisor) {
-                        left = exact_number<T>(); // 0
-                        right = tmp; // 1
-                    } else { // assuming D > 1. N > D ---> 1 < N / D < N
-                        left = tmp; // 1
-                        right = numerator;
-                    }
+                const T base = (std::numeric_limits<T>::max() / 4) * 2 - 1;
+                exact_number<T> half;
+                half.digits = {base / 2 + 1};
+
+                exact_number<T> left, right, numerator, denominator, residual, length;
```

Comment:
Its usually better to define in first use so you don't forget to initialize anything. If you trying to avoid writing exact_number<T> again and again, you can define a typedef at the start of the class and use it everytwhere.

Reply:
Ok, I'll change this.